src/profile-logic/profile-data.js

```diff
@@ -978,20 +979,45 @@ export function getTracingMarkers(thread: Thread): TracingMarker[] {
       };
       tracingMarkers.push(marker);
     } else if (data.type === 'tracing') {
+      // Tracing markers come in the form of 2 distinct markers: the first is
+      // emitted at the start of some process, the second at the end. We convert
+      // these 2 one-time markers into one marker with a non-null duration.
+      // We also handle nested markers by assuming markers of the same type are
+      // never interwoven: given input markers startA, startB, endC, endD, we'll
+      // get 2 markers A-D and B-C.
+      // This is implemented by AutoProfilerTracing in Gecko.
       const time = markers.time[i];
       const nameStringIndex = markers.name[i];
       if (data.interval === 'start') {
-        openMarkers.set(nameStringIndex, {
+        let markerBucket = openMarkers.get(nameStringIndex);
+        if (markerBucket === undefined) {
+          markerBucket = [];
+          openMarkers.set(nameStringIndex, markerBucket);
+        }
+
+        markerBucket.push({
           start: time,
           name: stringTable.getString(nameStringIndex),
           dur: 0,
           title: null,
           data,
         });
       } else if (data.interval === 'end') {
-        const marker = openMarkers.get(nameStringIndex);
-        if (marker === undefined) {
-          continue;
+        const markerBucket = openMarkers.get(nameStringIndex);
+        let marker;
+        if (markerBucket && markerBucket.length) {
+          marker = markerBucket.pop();
+        } else {
+          const nameStringIndex = markers.name[i];
+          // The tracing marker's start point is before the start of this
+          // profile. We'll miss the stack.
```

Comment:
I'm finding the phrasing here a bit confusing. Would you mind clarifying a bit?

Reply:
I used this, right after the `else`:
```
// No matching "start" marker has been encountered before this "end",
// this means it was issued before the capture started. Here we create
// a fake "start" marker to create the final tracing marker.
// Note we won't have additional data (eg the cause stack) for this
// marker because that data is contained in the "start" marker.
```